File: frontend/src/api/github.ts
import client from './client'

export interface GithubDailyStat {
  id: number
  user_id: number
  date: string
  commit_count: number
  pr_count: number
  issue_count: number
  star_delta: number
  created_at: string
  updated_at: string
}

export interface GithubDailyStatsResponse {
  data: GithubDailyStat[]
  total: number
}

export interface GithubSyncResponse {
  message: string
  repos_count: number
  stats_updated: number
  date_range: string
}

export interface SyncGithubParams {
  fromDate?: string
  toDate?: string
  mode?: 'standard' | 'deep'
}

/**
 * 同步 GitHub 数据（支持自定义时间范围）
 */
export async function syncGithubData(params?: SyncGithubParams): Promise<GithubSyncResponse> {
  const response = await client.post<GithubSyncResponse>('/github/sync', null, {
    params: {
      from_date: params?.fromDate,
      to_date: params?.toDate,
      mode: params?.mode,
    },
  })
  return response.data
}

/**
 * 获取每日统计数据
 */
export async function getDailyStats(
  fromDate?: string,
  toDate?: string
): Promise<GithubDailyStatsResponse> {
  const response = await client.get<GithubDailyStatsResponse>('/github/stats/daily', {
    params: {
      from_date: fromDate,
      to_date: toDate,
    },
  })
  return response.data
}

File: backend/app/schemas/github.py
"""
GitHub 相关的 Pydantic schemas
"""
from datetime import date, datetime
from typing import List, Optional

from pydantic import BaseModel


class GithubRepoBase(BaseModel):
    """GitHub 仓库基础信息"""

    repo_id: int
    name: str
    full_name: str
    private: bool
    language: Optional[str] = None
    html_url: str
    description: Optional[str] = None


class GithubRepoResponse(GithubRepoBase):
    """返回给客户端的仓库信息"""

    id: int
    user_id: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class GithubDailyStatBase(BaseModel):
    """GitHub 每日统计基础信息"""

    date: date
    commit_count: int = 0
    pr_count: int = 0
    issue_count: int = 0
    star_delta: int = 0


class GithubDailyStatResponse(GithubDailyStatBase):
    """返回给客户端的每日统计信息"""

    id: int
    user_id: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class GithubSyncResponse(BaseModel):
    """GitHub 数据同步响应"""

    message: str
    repos_count: int
    stats_updated: int
    date_range: Optional[str] = None


class GithubDailyStatsQueryResponse(BaseModel):
    """每日统计查询响应"""

    data: List[GithubDailyStatResponse]
    total: int

